src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,46 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        List<Machine> finalList = new ArrayList<>();
+        if (type == Bulldozer.class) {
+            BulldozerProducer bulldozer = new BulldozerProducer();
+            finalList.addAll(bulldozer.get());
+        }
+        if (type == Excavator.class) {
+            ExcavatorProducer excavator = new ExcavatorProducer();
+            finalList.addAll(excavator.get());
+        }
+        if (type == Truck.class) {
+            TruckProducer truck = new TruckProducer();
+            finalList.addAll(truck.get());
+        }
+
+        return finalList;
```

Comment:
Consider using an else-if structure instead of multiple if statements for checking the type. This will make the code more efficient and easier to read. According to the checklist: 'Keep if else constructions simple'.